sdk/registry.go

```diff
@@ -55,6 +55,11 @@ func (h *HookRegistry) Add(hook go_hook.GoHook) {
 	h.m.Lock()
 	defer h.m.Unlock()
 
+	config := hook.Config()
+	if config.OnStartup != nil && len(config.Kubernetes) > 0 {
+		panic("OnStartup and Kubernetes bindings cannot be used in a same hook")
```

Comment:
```suggestion
		panic("OnStartup hook always has binding context without Kubernetes snapshots. To prevent logic errors, don't use OnStartup and Kubernetes bindings in the same Go hook configuration.")
```